wpiutil/src/main/native/cpp/SafeThread.cpp

```diff
@@ -4,8 +4,37 @@
 
 #include "wpi/SafeThread.h"
 
+#include <atomic>
+
 using namespace wpi;
 
+// thread start/stop notifications for bindings that need to set up
+// per-thread state
+
+static void* DefaultOnThreadStart() {
+  return nullptr;
+}
+static void DefaultOnThreadEnd(void*) {}
+
+using OnThreadStartFn = void* (*)();
+using OnThreadEndFn = void (*)(void*);
+static wpi::mutex gSafeThreadNotifierLock;
+static int gSafeThreadRefcount;
```

Comment:
Just use an atomic here and remove the mutex.  The mutex may have static destruction ordering issues on Mac.